From fdad53465b428266779735ec32bfb9b83aec7834 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 21 Feb 2024 11:05:15 +0200
Subject: [PATCH] [SW-156330] gaudi3: add fns used to enable tpm spi chipsel

This patch defines functions used to initialize TPM
SPI chip select for gaudi3, these will be used
in common tpm files.

Change-Id: I832d10a0cb4cf45f3625eaeb39f431cadf551530
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index f7139f0..0d228d4 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -149,6 +149,14 @@
 	return D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_8);
 }
 
+void spi_ssi_cs1_get(unsigned int *mask, unsigned int *data_offset,
+		     unsigned int *out_en_offset)
+{
+	*mask = BIT(TPM_CS_GPIO_PIN % GPIOS_PER_BANK);
+	*data_offset = D0_PSOC_GPIO1_REG(mmGPIO_DATAOUT);
+	*out_en_offset = D0_PSOC_GPIO1_REG(mmGPIO_OUTENSET);
+}
+
 void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
 		     unsigned int *out_en_offset)
 {
diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index f53777e..c2bfe5a 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -100,4 +100,7 @@
 #define GAUDI3_NUM_HDCORES_PER_DIE	4
 #define GAUDI3_NUM_DCORES_PER_DIE	2
 
+#define GPIOS_PER_BANK			16
+#define TPM_CS_GPIO_PIN			18
+
 #endif /* COMMON_GAUDI3_ARC_GNU_H_ */
diff --git a/arch/gaudi3/common/common.c b/arch/gaudi3/common/common.c
index 9acb497..f9ee96c 100644
--- a/arch/gaudi3/common/common.c
+++ b/arch/gaudi3/common/common.c
@@ -6,6 +6,8 @@
 #include <baremetal.h>
 #include <hl.h>
 #include <gaudi3_blocks_linux_driver.h>
+#include <global_conf_regs.h>
+#include <spi_regs.h>
 #include <gaudi3_reg_map.h>
 #include <mstr_if_rr_lbw_regs.h>
 #include <pcie_core_regs.h>
@@ -125,3 +127,28 @@
 
 	return false;
 }
+
+void spi_tpm_base_get(unsigned int *regs)
+{
+	*regs = D0_PSOC_SPI_REG(mmSPI_CTRLR0);
+}
+
+unsigned int get_miso_pad_sel_tpm(void)
+{
+	return D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_15);
+}
+
+unsigned int get_mosi_pad_sel_tpm(void)
+{
+	return D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_16);
+}
+
+unsigned int get_sclk_out_pad_sel_tpm(void)
+{
+	return D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_17);
+}
+
+unsigned int get_cs_pad_sel_tpm(void)
+{
+	return D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PAD_SEL_18);
+}
